packages/codec/lib/format/utils/exception.ts

```diff
@@ -0,0 +1,147 @@
+import debugModule from "debug";
+const debug = debugModule("codec:utils:errors");
+
+import * as Format from "@truffle/codec/format/common";
+import * as Ast from "@truffle/codec/ast";
+import * as Storage from "@truffle/codec/storage/types";
+
+function isContractDefinedType(
```

Comment:
What is this function doing in this file??  It belongs with `isReferenceType` and etc.

Reply:
It's only used here and thus I decided its definition should live, unexported, in the only place where it's used.

Reply:
Yeah I still really think this should really go with `isReferenceType` and etc.  Putting it here is going to make it impossible to find later when it's needed somewhere else.  It does the same sort of thing as them and it has no logical connection to the point of this file.